controllers/grafana_controller.go

```diff
@@ -102,9 +102,7 @@ func (r *GrafanaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		nextStatus.StageStatus = status
 
 		if status != grafanav1beta1.OperatorStageResultSuccess {
-			controllerLog.Info("stage in progress", "stage", stage)
-			finished = false
-			break
+			return ctrl.Result{RequeueAfter: RequeueDelayError}, nil
```

Comment:
In that case the status would not get updated if a stage fails and the error message wouldn't be visible in the CR status.

Reply:
why return  here immediately instead of letting the status update happen?

Reply:
@pb82 it's something that I briefly mentioned during the last meeting. Not like I checked that, but I think status update will immediately trigger another reconciliation attempt (due to change passed through a WATCH session) instead of letting the operator wait for the `RequeueDelayError` period. WDYT?